javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -345,6 +345,18 @@ private int filter(Predicate<? super T> predicate, Object results, int index, T
         return index;
     }
 
+    Object[] toArray() {
+        final Object[] results = new Object[length];
+        visit((index, leaf, start, end) -> {
+            final int copied = end - start;
+            for (int i = start; i < end; i++) {
+                results[index + i] = type.getAt(leaf, i);
```

Comment:
> We observe that an iterator() is not as performant as a visitor (because of the additional method calls, see Vector).

the `visitor` is mainly needed to avoid boxing in case of primitive storage, it's not necessarily faster than the optimized iterator (that traverses leafs also).
If you think we need an alternative `visitor`/`iterator` also, I think it should be part of a separate issue, your previous suggestion solved the generalization problem, I think :)